python/tests/test_async_io.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2022-2024, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+
+import os
+
+import cupy
+import pytest
+
+import kvikio
+import kvikio.defaults
+
+
+def check_bit_flags(x: int, y: int) -> bool:
+    """Check that the bits set in `y` is also set in `x`"""
+    return x & y == y
+
+
+@pytest.mark.parametrize("size", [1, 10, 100, 1000, 1024, 4096, 4096 * 10])
+@pytest.mark.parametrize("nthreads", [1, 3, 4, 16])
+@pytest.mark.parametrize("tasksize", [199, 1024])
```

Comment:
Since `raw_read_async` and `raw_write_async` doesn't use the thread pool, we don't need `nthreads` and `tasksize`. 

Also, currently the C++ API's `read_async` and `write_async` doesn't use the `gds_threshold` argument.